### PR TITLE
Feat/introductory offer eligibility feature flag

### PR DESCRIPTION
**Pull request by hdahlheim**
Allow forcing the entry offer for everyone even if they had a membership already

### CODE REVIEW

**Review by ovbm — APPROVED** [review]
LGMT

File: packages/backend-modules/payments/graphql/resolvers/_mutations/createCheckoutSession.ts [hunk 1]
```diff
@@ -1,6 +1,10 @@
 /* eslint-disable @typescript-eslint/no-unused-vars */
 import { GraphqlContext } from '@orbiting/backend-modules-types'
-import { Shop, Offers, utils } from '../../../lib/shop'
+import {
+  Shop,
+  Offers,
+  checkIntroductoryOfferEligieblity,
```
Comment (ovbm) [inline packages/backend-modules/payments/graphql/resolvers/_mutations/createCheckoutSession.ts:6]: There's an "e" in the function name that shouldn't be there: checkIntroductoryOffer**Eligibility**

File: packages/backend-modules/payments/graphql/resolvers/_mutations/createCheckoutSession.ts [hunk 2]
```diff
@@ -25,8 +29,7 @@ export = async function createCheckoutSession(
   Auth.ensureUser(ctx.user)
 
   const shop = new Shop(Offers)
-  const entryOffer =
-    (await utils.hasHadMembership(ctx.user.id, ctx.pgdb)) === false
+  const entryOffer = await checkIntroductoryOfferEligieblity(ctx.pgdb, ctx.user)
```
Comment (jstcki) [inline packages/backend-modules/payments/graphql/resolvers/_mutations/createCheckoutSession.ts:32]: nit: this is named `withEntryOffer` or `withIntroductoryOffer` elsewhere, could update for consistency

File: packages/backend-modules/payments/graphql/resolvers/_queries/getOffers.ts [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
 
 export = async function getOffers(
   _root: never,
   args: { promoCode?: string },
   ctx: GraphqlContext,
 ) {
-  const entryOffer = ctx.user
-    ? (await utils.hasHadMembership(ctx?.user.id, ctx.pgdb)) === false
-    : true // if there is no user we show the entry offers
+  const entryOffer = await checkIntroductoryOfferEligieblity(ctx.pgdb, ctx.user)
```
Comment (jstcki) [inline packages/backend-modules/payments/graphql/resolvers/_queries/getOffers.ts:13]: `withIntroductoryOffer` also here?

File: packages/backend-modules/payments/graphql/resolvers/_queries/getOffer.ts [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
 export = async function getOffer(
   _root: never,
   args: { offerId: string; promoCode?: string },
   ctx: GraphqlContext,
 ) {
-  const withEntryOffer = ctx.user
-    ? (await utils.hasHadMembership(ctx.user.id, ctx.pgdb)) === false
-    : true // if there is no user we show the entry offers
-
+  const withEntryOffer = await checkIntroductoryOfferEligieblity(
```
Comment (jstcki) [inline packages/backend-modules/payments/graphql/resolvers/_queries/getOffer.ts:13]: `withIntroductoryOffer` also here?

**Review by jstcki — APPROVED** [review]
Looks good, just some naming things

File: packages/backend-modules/payments/graphql/resolvers/_queries/getOffer.ts (hunk 4 above)
Comment (hdahlheim) [reply packages/backend-modules/payments/graphql/resolvers/_queries/getOffer.ts:13]: I know all the places where it is inconsistent ;)
Will fix it.

**Pull request merged**